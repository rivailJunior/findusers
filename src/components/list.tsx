import React from 'react';
import { ErrorFeedback } from './errorFeedback';
import styles from './list.module.scss';
import moment from 'moment'

type iList = {
    data: object[],
    handleClick: Function
}

const ListRepositories = ({ data, handleClick }: iList): JSX.Element => {
    return data.length ? (
        <div data-testid="list" className={styles.item}>
            {data.map((item: any, index) => {
                return (
                    <div key={index} className={styles.itemLi} onClick={() => handleClick(item)} data-testid="listItem">
                        <div className={styles.left}>
                            {item.stargazers_count}
                        </div>
                        <div className={styles.right}>
                            <div className={styles.rightTitle}>
                                {item?.name}
                            </div>
                            <div className={styles.rightSub}>
                                <div>{moment(item.created_at).format('DD/MM/YYYY')}</div>
                                <div> <a href={item.clone_url} target="_blank" rel="noopener noreferrer">{item.clone_url}</a></div>
                            </div>
                        </div>
                    </div>)
            })}
        </div>
    ) : (
        <ErrorFeedback title="Nada encontrado" />
    );
}

export { ListRepositories };
