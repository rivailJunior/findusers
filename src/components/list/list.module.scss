@use '../../pages/index.scss' as index;
@use '../../pages/root.scss' as root;
.item {
    display: flex;
    flex-direction: column;
    border-radius: 5px;

    .itemLi {
        @include index.default-label(15px, 0, 100%, 300, 15px);
        display: grid;
        text-align: left;
        padding: 1%;
        grid-template-columns: repeat(12, 1fr);
        max-height: 100px;
        background: #ffffff 0% 0% no-repeat padding-box;
        border-radius: 4px;
        opacity: 1;
        margin-bottom: 3px;
        max-height: 80px;

        .left {
            grid-column: span 2;
            display: flex;
            justify-content: center;
            align-self: center;
            font-size: 3rem;
        }
        .right {
            grid-column: span 10;

            .rightTitle {
                @include index.default-label(15px, 0, 100%, 700, 20px);
                text-transform: capitalize;
                text-align: left;
            }
            .rightSub {
                @include index.default-label(15px, 0px, 100%, 300, 20px, left);
                display: flex;
                flex-direction: column;
                justify-content: space-between;
                div {
                    width: 100%;
                    overflow: hidden;
                    text-overflow: ellipsis;
                }
            }
        }
    }
    .itemLi:hover {
        cursor: pointer;
        background-color: root.$hoverBgc;
        @include index.default-shadow();
    }
}

.listTitlesInformations {
    display: grid;
    grid-template-columns: repeat(12, 1fr);
    background-color: white;
    padding: 1%;
    border-bottom: 1px solid;

    .button {
        display: flex;
        justify-content: center;
        grid-column: span 2;
    }
    .description {
        grid-column: span 8;
    }
    .stars {
        grid-column: span 2;
    }
}

.listReverse {
    display: flex;
    flex-direction: column-reverse;
}

.btnReverse {
    background-color: root.$btnBgc;
    color: white;
    height: 20px;
    width: 40px;
    /* box-shadow: 0px 0px 5px #00000033; */
    border-radius: 4px;
    border-style: none;
    cursor: pointer;
}

/* Smartphones (portrait and landscape) ----------- */
@media only screen and (max-device-width: 320px) and (max-device-width: 480px) {
    .item {
        .itemLi {
            grid-template-columns: repeat(12, 1fr);
            margin-bottom: 3px;

            .left {
                display: none;
            }

            .right {
                grid-column: span 12;

                .rightTitle {
                    @include index.default-label(13px, 1%, 100%, 500, 10px, left);
                    text-transform: capitalize;
                }
                .rightSub {
                    @include index.default-label(12px, 5px, 100%, 300, 17px, left);
                    display: flex;
                    flex-direction: column;
                    justify-content: space-between;
                    overflow: hidden;
                }
            }
        }
        .itemLi:hover {
            cursor: pointer;
            color: root.$hoverColor;
        }
    }

    .listTitlesInformations {
        padding: 1%;

        .button {
            grid-column: span 2;
        }
        .description {
            grid-column: span 10;
        }
        .stars {
            display: none;
        }
    }
}

/* iPads (portrait and landscape) ----------- */
@media only screen and (min-device-width: 481px) and (max-device-width: 1024px) {
    .item {
        .itemLi {
            grid-template-columns: repeat(12, 1fr);
            margin-bottom: 3px;

            .left {
                display: none;
            }

            .right {
                grid-column: span 12;

                .rightTitle {
                    @include index.default-label(13px, 1%, 100%, 500, 10px, left);
                    text-transform: capitalize;
                }
                .rightSub {
                    @include index.default-label(12px, 5px, 100%, 300, 17px, left);
                    display: flex;
                    flex-direction: column;
                    justify-content: space-between;
                    overflow: hidden;
                    width: 100%;
                }
            }
        }
        .itemLi:hover {
            cursor: pointer;
            color: root.$hoverColor;
        }
    }
}

/* Desktops and laptops ----------- */
@media only screen and (min-width: 1224px) {
    /* Styles */
}
