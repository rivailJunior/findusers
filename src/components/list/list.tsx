import React, { useState } from 'react';
import { ErrorFeedback } from '../errorFeedback/errorFeedback';
import styles from './list.module.scss';
import moment from 'moment'
import { GridContainer, GridContainerItem } from '../gridContainer/gridContainer';


type iList = {
    data: object[],
    handleClick: Function
}

const ListRepositories = ({ data, handleClick }: iList): JSX.Element => {
    const [showReverse, setShowReverse] = useState(false);

    const listStyle = showReverse ? `${ styles.item } ${ styles.listReverse }` : styles.item;
    return data.length ? (
        <div>
            <div className={styles.listTitlesInformations}>
                <div className={styles.stars}>
                    <label htmlFor="">Estrelas</label>
                </div>
                <div className={styles.description}>
                    <label htmlFor="">Description</label>
                </div>
                <div className={styles.button}>
                    <button data-testid="btnReverse" className={styles.btnReverse} onClick={() => setShowReverse(!showReverse)}>Sort</button>
                </div>
            </div>


            <div data-testid="list" className={listStyle}>

                {data.map((item: any, index) => {
                    return (
                        <div key={index} className={styles.itemLi} onClick={() => handleClick(item)} data-testid="listItem">
                            <div className={styles.left}>
                                {item.stargazers_count}
                            </div>
                            <div className={styles.right}>
                                <div className={styles.rightTitle}>
                                    {item?.name}
                                </div>
                                <div className={styles.rightSub}>
                                    <div>{moment(item.created_at).format('DD/MM/YYYY')}</div>
                                    <div> <a href={item.clone_url} target="_blank" rel="noopener noreferrer">{item.clone_url}</a></div>
                                </div>
                            </div>
                        </div>)
                })}
            </div>
        </div>
    ) : (
        <ErrorFeedback title="Nada encontrado" />
    );
}

export { ListRepositories };
