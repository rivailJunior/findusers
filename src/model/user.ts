export const UserValue: User = {
    avatar_url: "https://avatars.githubusercontent.com/u/5783143?v=4",
    bio: "Full Stack developer, javascript lover and an open source enthusiastic.",
    blog: "",
    company: "Qtarkuss",
    created_at: "2013-10-26T19:17:03Z",
    email: "rivail.rj@gmail.com",
    events_url: "https://api.github.com/users/rivailJunior/events{/privacy}",
    followers: 4,
    followers_url: "https://api.github.com/users/rivailJunior/followers",
    following: 2,
    following_url: "https://api.github.com/users/rivailJunior/following{/other_user}",
    gists_url: "https://api.github.com/users/rivailJunior/gists{/gist_id}",
    gravatar_id: "",
    hireable: true,
    html_url: "https://github.com/rivailJunior",
    id: 0,
    location: "Brasil",
    login: "rivailJunior",
    name: "Rivail Santos",
    node_id: "MDQ6VXNlcjU3ODMxNDM=",
    organizations_url: "https://api.github.com/users/rivailJunior/orgs",
    public_gists: 0,
    public_repos: 12,
    received_events_url: "https://api.github.com/users/rivailJunior/received_events",
    repos_url: "https://api.github.com/users/rivailJunior/repos",
    site_admin: false,
    starred_url: "https://api.github.com/users/rivailJunior/starred{/owner}{/repo}",
    subscriptions_url: "https://api.github.com/users/rivailJunior/subscriptions",
    twitter_username: null,
    type: "User",
    updated_at: "2021-04-25T03:20:04Z",
    url: "https://api.github.com/users/rivailJunior"
}

export type User = {
    avatar_url: string
    bio: string
    blog: string
    company: string
    created_at: string
    email: string
    events_url: string
    followers: number
    followers_url: string
    following: number
    following_url: string
    gists_url: string
    gravatar_id: string
    hireable: boolean
    html_url: string
    id: number
    location: string
    login: string
    name: string
    node_id: string
    organizations_url: string
    public_gists: number
    public_repos: number
    received_events_url: string
    repos_url: string
    site_admin: false
    starred_url: string
    subscriptions_url: string
    twitter_username: null
    type: string
    updated_at: string
    url: string
}