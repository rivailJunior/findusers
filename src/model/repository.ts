import { User, UserValue } from './user';
export type Repository = {
    archive_url: string
    archived: boolean
    assignees_url: string
    blobs_url: string
    branches_url: string
    clone_url: string
    collaborators_url: string
    comments_url: string
    commits_url: string
    compare_url: string
    contents_url: string
    contributors_url: string
    created_at: string
    default_branch: string
    deployments_url: string
    description: any
    disabled: boolean
    downloads_url: string
    events_url: string
    fork: number
    forks: number
    forks_count: number
    forks_url: string
    full_name: string
    git_commits_url: string
    git_refs_url: string
    git_tags_url: string
    git_url: string
    has_downloads: boolean
    has_issues: boolean
    has_pages: boolean
    has_projects: boolean
    has_wiki: boolean
    homepage: any
    hooks_url: string
    html_url: string
    id: number
    issue_comment_url: string
    issue_events_url: string
    issues_url: string
    keys_url: string
    labels_url: string
    language: string
    languages_url: string
    license: { key: string, name: string, spdx_id: string, url: string }
    merges_url: string
    milestones_url: string
    mirror_url: any
    name: string
    network_count: number
    node_id: string
    notifications_url: string
    open_issues: number
    open_issues_count: number
    owner: User
    permissions: Object
    private: false
    pulls_url: string
    pushed_at: string
    releases_url: string
    size: number
    ssh_url: string
    stargazers_count: number
    stargazers_url: string
    statuses_url: string
    subscribers_count: number
    subscribers_url: string
    subscription_url: string
    svn_url: string
    tags_url: string
    teams_url: string
    temp_clone_token: string
    trees_url: string
    updated_at: string
    url: string
    watchers: number
    watchers_count: number
}

export const repositoryMockValue = {
    archive_url: "https://api.github.com/repos/rivailJunior/moveon/{archive_format}{/ref}",
    archived: false,
    assignees_url: "https://api.github.com/repos/rivailJunior/moveon/assignees{/user}",
    blobs_url: "https://api.github.com/repos/rivailJunior/moveon/git/blobs{/sha}",
    branches_url: "https://api.github.com/repos/rivailJunior/moveon/branches{/branch}",
    clone_url: "https://github.com/rivailJunior/moveon.git",
    collaborators_url: "https://api.github.com/repos/rivailJunior/moveon/collaborators{/collaborator}",
    comments_url: "https://api.github.com/repos/rivailJunior/moveon/comments{/number}",
    commits_url: "https://api.github.com/repos/rivailJunior/moveon/commits{/sha}",
    compare_url: "https://api.github.com/repos/rivailJunior/moveon/compare/{base}...{head}",
    contents_url: "https://api.github.com/repos/rivailJunior/moveon/contents/{+path}",
    contributors_url: "https://api.github.com/repos/rivailJunior/moveon/contributors",
    created_at: "2016-06-02T19:09:35Z",
    default_branch: "master",
    deployments_url: "https://api.github.com/repos/rivailJunior/moveon/deployments",
    description: null,
    disabled: false,
    downloads_url: "https://api.github.com/repos/rivailJunior/moveon/downloads",
    events_url: "https://api.github.com/repos/rivailJunior/moveon/events",
    fork: false,
    forks: 0,
    forks_count: 0,
    forks_url: "https://api.github.com/repos/rivailJunior/moveon/forks",
    full_name: "rivailJunior/moveon",
    git_commits_url: "https://api.github.com/repos/rivailJunior/moveon/git/commits{/sha}",
    git_refs_url: "https://api.github.com/repos/rivailJunior/moveon/git/refs{/sha}",
    git_tags_url: "https://api.github.com/repos/rivailJunior/moveon/git/tags{/sha}",
    git_url: "git://github.com/rivailJunior/moveon.git",
    has_downloads: true,
    has_issues: true,
    has_pages: false,
    has_projects: true,
    has_wiki: true,
    homepage: null,
    hooks_url: "https://api.github.com/repos/rivailJunior/moveon/hooks",
    html_url: "https://github.com/rivailJunior/moveon",
    id: 60290361,
    issue_comment_url: "https://api.github.com/repos/rivailJunior/moveon/issues/comments{/number}",
    issue_events_url: "https://api.github.com/repos/rivailJunior/moveon/issues/events{/number}",
    issues_url: "https://api.github.com/repos/rivailJunior/moveon/issues{/number}",
    keys_url: "https://api.github.com/repos/rivailJunior/moveon/keys{/key_id}",
    labels_url: "https://api.github.com/repos/rivailJunior/moveon/labels{/name}",
    language: "HTML",
    languages_url: "https://api.github.com/repos/rivailJunior/moveon/languages",
    license: { key: "mit", name: "MIT License", spdx_id: "MIT", url: "https://api.github.com/licenses/mit" },
    merges_url: "https://api.github.com/repos/rivailJunior/moveon/merges",
    milestones_url: "https://api.github.com/repos/rivailJunior/moveon/milestones{/number}",
    mirror_url: null,
    name: "moveon",
    network_count: 0,
    node_id: "MDEwOlJlcG9zaXRvcnk2MDI5MDM2MQ==",
    notifications_url: "https://api.github.com/repos/rivailJunior/moveon/notifications{?since,all,participating}",
    open_issues: 7,
    open_issues_count: 7,
    owner: UserValue,
    permissions: { admin: true, push: true, pull: true },
    private: false,
    pulls_url: "https://api.github.com/repos/rivailJunior/moveon/pulls{/number}",
    pushed_at: "2016-07-19T19:56:10Z",
    releases_url: "https://api.github.com/repos/rivailJunior/moveon/releases{/id}",
    size: 14671,
    ssh_url: "git@github.com:rivailJunior/moveon.git",
    stargazers_count: 1,
    stargazers_url: "https://api.github.com/repos/rivailJunior/moveon/stargazers",
    statuses_url: "https://api.github.com/repos/rivailJunior/moveon/statuses/{sha}",
    subscribers_count: 5,
    subscribers_url: "https://api.github.com/repos/rivailJunior/moveon/subscribers",
    subscription_url: "https://api.github.com/repos/rivailJunior/moveon/subscription",
    svn_url: "https://github.com/rivailJunior/moveon",
    tags_url: "https://api.github.com/repos/rivailJunior/moveon/tags",
    teams_url: "https://api.github.com/repos/rivailJunior/moveon/teams",
    temp_clone_token: "",
    trees_url: "https://api.github.com/repos/rivailJunior/moveon/git/trees{/sha}",
    updated_at: "2016-09-20T14:47:58Z",
    url: "https://api.github.com/repos/rivailJunior/moveon",
    watchers: 1,
    watchers_count: 1
}