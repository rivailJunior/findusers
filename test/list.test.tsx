import React from 'react'
import { render, screen, fireEvent, act } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { ListRepositories } from '../src/components/list/list'
import { repositoryMockValue } from '../src/model/repository'

import { unmountComponentAtNode } from "react-dom";

let container = null;
beforeEach(() => {
    // configurar o elemento do DOM como o alvo da renderização
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // limpar na saída
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

const arrayFake = Array(10).fill(repositoryMockValue)

const fakeFn = jest.fn()
describe('List Component', () => {
    test('Render Correctly', () => {
        const container = renderer.create(<ListRepositories data={arrayFake} handleClick={fakeFn} />)
        container.toJSON();
        expect(container).toMatchSnapshot()
    })

    test('Show correct total itens on list', () => {
        const { getAllByTestId } = render(<ListRepositories data={arrayFake} handleClick={fakeFn} />, container)
        const listLi = getAllByTestId('listItem')
        expect(listLi).toHaveLength(arrayFake.length)
    })

    test('Click on list item', () => {
        const { getAllByTestId } = render(<ListRepositories data={arrayFake} handleClick={fakeFn} />, container)
        const listLi = getAllByTestId('listItem')
        fireEvent.click(listLi[0])
        expect(fakeFn).toHaveBeenCalledTimes(1)
    })


    test('Click on sort', (done) => {
        render(<ListRepositories data={arrayFake} handleClick={fakeFn} />, container)
        const list = screen.getByTestId('list')
        const btnReverse = screen.getByTestId("btnReverse");
        expect(list.className).toBe("item");
        fireEvent.click(btnReverse)
        const sortedList = screen.getByTestId('list')
        expect(sortedList.className).toBe("item listReverse")
        done()

    })

})