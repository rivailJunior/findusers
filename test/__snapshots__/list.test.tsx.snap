// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component Render Correctly 1`] = `
<div>
  <div
    className="listTitlesInformations"
  >
    <div
      className="stars"
    >
      <label
        htmlFor=""
      >
        Estrelas
      </label>
    </div>
    <div
      className="description"
    >
      <label
        htmlFor=""
      >
        Description
      </label>
    </div>
    <div
      className="button"
    >
      <button
        className="btnReverse"
        data-testid="btnReverse"
        onClick={[Function]}
      >
        Sort
      </button>
    </div>
  </div>
  <div
    className="item"
    data-testid="list"
  >
    <div
      className="itemLi"
      data-testid="listItem"
      onClick={[Function]}
    >
      <div
        className="left"
      >
        1
      </div>
      <div
        className="right"
      >
        <div
          className="rightTitle"
        >
          moveon
        </div>
        <div
          className="rightSub"
        >
          <div>
            02/06/2016
          </div>
          <div>
             
            <a
              href="https://github.com/rivailJunior/moveon.git"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://github.com/rivailJunior/moveon.git
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="itemLi"
      data-testid="listItem"
      onClick={[Function]}
    >
      <div
        className="left"
      >
        1
      </div>
      <div
        className="right"
      >
        <div
          className="rightTitle"
        >
          moveon
        </div>
        <div
          className="rightSub"
        >
          <div>
            02/06/2016
          </div>
          <div>
             
            <a
              href="https://github.com/rivailJunior/moveon.git"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://github.com/rivailJunior/moveon.git
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="itemLi"
      data-testid="listItem"
      onClick={[Function]}
    >
      <div
        className="left"
      >
        1
      </div>
      <div
        className="right"
      >
        <div
          className="rightTitle"
        >
          moveon
        </div>
        <div
          className="rightSub"
        >
          <div>
            02/06/2016
          </div>
          <div>
             
            <a
              href="https://github.com/rivailJunior/moveon.git"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://github.com/rivailJunior/moveon.git
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="itemLi"
      data-testid="listItem"
      onClick={[Function]}
    >
      <div
        className="left"
      >
        1
      </div>
      <div
        className="right"
      >
        <div
          className="rightTitle"
        >
          moveon
        </div>
        <div
          className="rightSub"
        >
          <div>
            02/06/2016
          </div>
          <div>
             
            <a
              href="https://github.com/rivailJunior/moveon.git"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://github.com/rivailJunior/moveon.git
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="itemLi"
      data-testid="listItem"
      onClick={[Function]}
    >
      <div
        className="left"
      >
        1
      </div>
      <div
        className="right"
      >
        <div
          className="rightTitle"
        >
          moveon
        </div>
        <div
          className="rightSub"
        >
          <div>
            02/06/2016
          </div>
          <div>
             
            <a
              href="https://github.com/rivailJunior/moveon.git"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://github.com/rivailJunior/moveon.git
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="itemLi"
      data-testid="listItem"
      onClick={[Function]}
    >
      <div
        className="left"
      >
        1
      </div>
      <div
        className="right"
      >
        <div
          className="rightTitle"
        >
          moveon
        </div>
        <div
          className="rightSub"
        >
          <div>
            02/06/2016
          </div>
          <div>
             
            <a
              href="https://github.com/rivailJunior/moveon.git"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://github.com/rivailJunior/moveon.git
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="itemLi"
      data-testid="listItem"
      onClick={[Function]}
    >
      <div
        className="left"
      >
        1
      </div>
      <div
        className="right"
      >
        <div
          className="rightTitle"
        >
          moveon
        </div>
        <div
          className="rightSub"
        >
          <div>
            02/06/2016
          </div>
          <div>
             
            <a
              href="https://github.com/rivailJunior/moveon.git"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://github.com/rivailJunior/moveon.git
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="itemLi"
      data-testid="listItem"
      onClick={[Function]}
    >
      <div
        className="left"
      >
        1
      </div>
      <div
        className="right"
      >
        <div
          className="rightTitle"
        >
          moveon
        </div>
        <div
          className="rightSub"
        >
          <div>
            02/06/2016
          </div>
          <div>
             
            <a
              href="https://github.com/rivailJunior/moveon.git"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://github.com/rivailJunior/moveon.git
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="itemLi"
      data-testid="listItem"
      onClick={[Function]}
    >
      <div
        className="left"
      >
        1
      </div>
      <div
        className="right"
      >
        <div
          className="rightTitle"
        >
          moveon
        </div>
        <div
          className="rightSub"
        >
          <div>
            02/06/2016
          </div>
          <div>
             
            <a
              href="https://github.com/rivailJunior/moveon.git"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://github.com/rivailJunior/moveon.git
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="itemLi"
      data-testid="listItem"
      onClick={[Function]}
    >
      <div
        className="left"
      >
        1
      </div>
      <div
        className="right"
      >
        <div
          className="rightTitle"
        >
          moveon
        </div>
        <div
          className="rightSub"
        >
          <div>
            02/06/2016
          </div>
          <div>
             
            <a
              href="https://github.com/rivailJunior/moveon.git"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://github.com/rivailJunior/moveon.git
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
